### DISCUSSION TITLE
Max delegate is dangerous!

### DISCUSSION THREAD

**Discussion by dbwebapp** [BitCannaGlobal/WeedWallet#122] [Ideas]
![Screenshot 2023-06-07 at 22 01 15](https://github.com/BitCannaGlobal/WeedWallet/assets/96944913/69857ede-0b12-4ce3-b67f-6a2ac7f05795)

When clicked on button MAX when delegating, it should automatically create a tx but with some random % less of tokens to still have for gas fees to avoid this situation : 

![Screenshot 2023-06-07 at 21 59 41](https://github.com/BitCannaGlobal/WeedWallet/assets/96944913/692d65af-eed7-40fc-9f80-f8c6310d6641)

**Comment by dbwebapp** [comment]
Max button should be like "I want to delegate as much as possible " so it should not  be a situation when you have to go back and manually write the maximum to delegate excluded the gas fees. In the best cases, we keep at least gas fees for 2 tx, 1 to delegate MAX and 1 for claim rewards , since after claiming rewards fresh tokens are getting on the wallet ...

**Comment by TJ-BitCanna** [comment]
Current setup already has a substraction of the transaction fees to perform the transaction :)
